### PR TITLE
optimizes icon2html() for icon files known to be in the rsc at compile time

### PR DESCRIPTION
**Pull request by triplezeta**
its just a direct port of https://github.com/tgstation/tgstation/pull/67429

## why its good for deez nuts
read that PR and see why. Also it makes both the context menu and the alt click statpanel menu way better from what I saw in localhost testing.

### CODE REVIEW

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -51,6 +51,8 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define NOJAUNT_1 (1<<0)
 /// If a turf can be made dirty at roundstart. This is also used in areas.
 #define CAN_BE_DIRTY_1 (1<<1)
+/// Should we use the initial icon for display? Mostly used by overlay only objects
+#define HTML_USE_INITAL_ICON_1 (1<<20)
```
Comment (Fikou) [inline code/__DEFINES/flags.dm]: you don't seem to have added this to anything

Comment (Fikou) [reply code/__DEFINES/flags.dm]: also why is this in turf flags

Comment (triplezeta) [reply code/__DEFINES/flags.dm]: pasted from TG to make the code compile

**Review by ZephyrTFA — APPROVED** [review]
I had asked for this to be ported; it looks like it was delegated to Tetra.
This is approved, but I suggest, and require, it be test merged to ensure it doesn't secretly break everything

**Comment by triplezeta** [comment]
i wouldnt say delegated

File: code/__DEFINES/flags.dm (hunk 1 above)
Comment (Fikou) [reply code/__DEFINES/flags.dm]: move this to be where it should
and add it to places where it is on tg

Comment (triplezeta) [reply code/__DEFINES/flags.dm]: leaving this comment as a reminder to myself to actually do this

Comment (triplezeta) [reply code/__DEFINES/flags.dm]: ok done

**Comment by triplezeta** [comment]
`HTML_USE_INITAL_ICON_1` moved out of turf flags and added to things that need it.

In localhost, this appears to make the alt-click stat panel menu and context menus both significantly more accurate with icons.

**Pull request merged**